### PR TITLE
feat: add SearchInput component

### PR DESCRIPTION
**Pull request by snitin315**
## Description

This PR adds a new component - `SearchInput`.

## Changes

- API - https://github.com/razorpay/blade/pull/2139
- Tests - https://github.com/razorpay/blade/pull/2144

## Component Checklist

- [ ] Update Component Status Page
- [x] Perform Manual Testing in Other Browsers
- [ ] Add KitchenSink Story
- [x] Add Interaction Tests (if applicable)
- [x] Add changeset

### CODE REVIEW

File: packages/blade/src/components/Input/SearchInput/SearchInput.tsx [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+  const textInputRef = React.useRef<BladeElementRefWithValue>(null);
+  const mergedRef = useMergeRefs(ref, textInputRef);
+  const [shouldShowClearButton, setShouldShowClearButton] = useState(false);
+  const {
+    triggererWrapperRef,
+    onTriggerKeydown,
+    onTriggerClick,
+    dropdownTriggerer,
+  } = useDropdown();
+  const isDropdownUsed = dropdownTriggerer === 'SearchInput';
```
Comment (anuraghazra) [inline packages/blade/src/components/Input/SearchInput/SearchInput.tsx:132]:
```suggestion
  const isInsideDropdown = dropdownTriggerer === 'SearchInput';
```

**Review by anuraghazra — APPROVED** [review]

File: packages/blade/src/components/ActionList/ActionListItem.tsx [hunk 2]
```diff
@@ -31,7 +31,7 @@ import { dropdownComponentIds } from '~components/Dropdown/dropdownComponentIds'
 type ActionListItemProps = {
   title: string;
   description?: string;
-  onClick?: (clickProps: { name: string; value?: boolean }) => void;
+  onClick?: (clickProps: { name: string; value?: boolean; event: React.MouseEvent }) => void;
```
Comment (saurabhdaware) [inline packages/blade/src/components/ActionList/ActionListItem.tsx]: Handle this type for React Native

**Review by saurabhdaware — APPROVED** [review]

File: packages/blade/src/components/Input/SearchInput/SearchInput.stories.tsx
Comment (saurabhdaware) [inline packages/blade/src/components/Input/SearchInput/SearchInput.stories.tsx:1]: 1. Can you add some basic filtering in SearchInput documentation? 
2. Try getting ActionList info from some sample API or simulate getting data after delay. ActionList might work a bit differently and strangely in that case so better to have that story https://61c19ee8d3d282003ac1d81c-kmpprqittc.chromatic.com/?path=/story/components-input-searchinput--search-input-with-dropdown

Comment (snitin315) [reply packages/blade/src/components/Input/SearchInput/SearchInput.stories.tsx:1]: done.

**Review by anuraghazra — APPROVED** [review]

**Pull request merged**